flow/model/cdc_stream.go

```diff
@@ -81,23 +85,27 @@ func (r *CDCStream[T]) AddRecord(ctx context.Context, record Record[T]) error {
 }
 
 func (r *CDCStream[T]) SignalAsEmpty() {
-	r.emptySignal <- true
+	r.emptySet = true
+	close(r.emptySignal)
 }
 
 func (r *CDCStream[T]) SignalAsNotEmpty() {
-	r.emptySignal <- false
+	r.empty = false
+	r.emptySet = true
```

Comment:
can we make `empty` a `*bool` and have nil, false and true be the values instead? or some other form of tristate variable
having 2 of them is unnecessary

Reply:
I'd rather keep 2 bools without pointer indirection. Could make it a single byte which we pull bits out of, but that muddies things, & with alignment there's no impact on struct size